src/finding/summary.go

```diff
@@ -137,3 +164,56 @@ func findingAttributeValue(finding types.ImageScanFinding, name string) string {
 	}
 	return ""
 }
+
+const legacyCVEURL = "https://cve.mitre.org/cgi-bin/cvename.cgi?name="
+const updatedCVEURL = "https://www.cve.org/CVERecord?id="
+
+func fixFindingURI(name string, uri string) string {
+	correctedURI := uri
+
+	// transition from the old CVE site that is deprecated
+	if strings.HasPrefix(correctedURI, legacyCVEURL) {
+		correctedURI = strings.Replace(correctedURI, legacyCVEURL, updatedCVEURL, 1)
+	}
+
+	// sometimes links are published that are not valid: in this case point to a
+	// GH vuln search as a way to provide some value
+	if strings.HasPrefix(correctedURI, updatedCVEURL) && !strings.HasPrefix(name, "CVE-") {
+		correctedURI = "https://github.com/advisories?query=" + url.QueryEscape(name)
+	}
+
+	return correctedURI
+}
+
+func cvss2VectorURL(cvss2Vector string) string {
+	cvss2VectorURL := ""
+	if cvss2Vector != "" {
+		cvss2VectorURL = "https://nvd.nist.gov/vuln-metrics/cvss/v2-calculator?vector=" +
+			url.QueryEscape("("+cvss2Vector+")")
+	}
+	return cvss2VectorURL
+}
```

Comment:
Ref https://github.com/lindell/multi-gitter/pull/397#discussion_r1372144001

The suggestion was that it's clear without context what is occurring. There's no real advantage in either.

Reply:
```suggestion
func cvss2VectorURL(cvss2Vector string) string {
	if cvss2Vector == "" {
		return ""
	}

	return "https://nvd.nist.gov/vuln-metrics/cvss/v2-calculator?vector=" +
		url.QueryEscape("("+cvss2Vector+")")
}
```

Gonna use the early return (returning the empty string), but avoid the `const`

Reply:
In Go's standard libraries that is a pattern. Otherwise:

`if len(s) = 0`

Reply:
This is a good SO answer: https://stackoverflow.com/a/18595217/138653

Seems like there is a bias towards `== ""` but with a caveat it might be clearer to use `len(s)` if the context includes string indices.

Reply:
It's interesting: across languages I've tended towards the equivalent of `len(s)`, but I hadn't considered that Go's `len()` is used for multiple types. This did change my thinking and my reading prompted by the above seems to back up a shift towards `== ""`. It doesn't seem like a "hard" shift, more of a preference.

Reply:
lgtm. Why not use the constant? It's (arguably) more readable and removed/inlined by the compiler.

> ...it might be clearer to use len(s) if the context includes string indices.

Agreed, and I think `s == ""` is more readable when checking for empty-strings only. (It's very clear what's happening.)

Like you, I default to a `len`/`length` comparison due to other languages, which is why I'm curious about the _preferred_ method in Go

Reply:
It's a trivial detail and either solution is acceptable. Just curious about your reasoning